libraries/DataFlash/LogStructure.h

```diff
@@ -34,6 +36,91 @@ struct PACKED log_Format {
     char labels[64];
 };
 
+struct PACKED log_Unit {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    char unit[64]; // you know, this might be overkill...
+};
+
+struct PACKED log_Format_Multiplier {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    double multiplier;
+};
+
+struct PACKED log_Format_Units {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t format_type;
+    char units[16];
+    char multipliers[16];
+};
+
+struct UnitStructure {
+    const char ID;
+    const char unit[64];
+};
+
+struct MultiplierStructure {
+    const char ID;
+    const double multiplier;
+};
+
+// all units here should be base units
+// This does mean battery capacity is here as "amps*seconds"
+const struct UnitStructure log_Units[] = {
+    { '-', "" },    // no units e.g. Pi, or a string
+    { '?', "UNKNOWN" },    // Units which haven't been worked out yet....
+    { 'A', "amps" },
```

Comment:
I LOVE this PR, but can you please use the same units as in Tools/autotest/param_metadata/param.py:44 ?

Reply:
So basically you're suggesting rolling in the multiplier and the unit into one field....

What happens when that cross product exceeds ~100 bytes? (the printable ASCII characters).  The cross-product effect is nasty.

It is critical that the textual unit be as parsable as possible for the ground control stations.  Any time they hit a unit they can't parse (e.g. "microsecond"), they can't manipulate the values any further (I'm thinking graphing here).  Splitting the multiplier off will reduce the code changes required on ground control stations when we add a unit.  I've made an effort to use base-SI units where available in the units for this reason.

Reply:
@peterbarker I'd say the text (long form of the name) should probably match the sane unit over there, but after that I would not be in favour of the cross product version.

Reply:
No I meant using "ampere" instead of "amps" https://github.com/ArduPilot/ardupilot/blob/master/Tools/autotest/param_metadata/param.py#L74
"volt" instead of "volts"
'degrees per second' instead of "degrees/second".
"ampere second" instead of "amps*seconds"
and so on. I also made an eford to use SI units and read a lot of documents do do that. the links are here:
https://github.com/ArduPilot/ardupilot/blob/master/Tools/autotest/param_metadata/param.py#L33

I'm all about the separated multipliers, so do not change that, please. :)

Reply:
@amilcarlucas Those notes are awesome, thanks!  I believe I've changed them all to the canonical SI names.